### PR TITLE
nixos: shellhub-agent: support more options

### PR DESCRIPTION
**Pull request by otavio**
###### Description of changes

Rework of existing module and addition of new options.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/networking/shellhub-agent.nix [hunk 1]
```diff
@@ -74,9 +89,11 @@ in {
...
       ];
 
-      environment.SERVER_ADDRESS = cfg.server;
-      environment.PRIVATE_KEY = cfg.privateKey;
-      environment.TENANT_ID = cfg.tenantId;
+      environment.SHELLHUB_SERVER_ADDRESS = cfg.server;
+      environment.SHELLHUB_PRIVATE_KEY = cfg.privateKey;
+      environment.SHELLHUB_TENANT_ID = cfg.tenantId;
+      environment.SHELLHUB_PREFERRED_HOSTNAME = cfg.preferredHostname;
+      environment.SHELLHUB_KEEPALIVE_INTERVAL = toString cfg.keepAliveInterval;
```
Comment (SuperSandro2000) [inline nixos/modules/services/networking/shellhub-agent.nix:96]:
```suggestion
      environment = {
        SHELLHUB_SERVER_ADDRESS = cfg.server;
        SHELLHUB_PRIVATE_KEY = cfg.privateKey;
        SHELLHUB_TENANT_ID = cfg.tenantId;
        SHELLHUB_PREFERRED_HOSTNAME = cfg.preferredHostname;
        SHELLHUB_KEEPALIVE_INTERVAL = toString cfg.keepAliveInterval;
      };
```
We could reduce repetition of environment if you like.

**Review by SuperSandro2000 — COMMENTED** [review]
The commit should start with ``nixos/shellhub-agent``.

File: nixos/modules/services/networking/shellhub-agent.nix (hunk 1 above)
Comment (otavio) [reply nixos/modules/services/networking/shellhub-agent.nix:96]: Sure. I'll do it here and send the update.

File: nixos/modules/services/networking/shellhub-agent.nix [hunk 2]
```diff
@@ -1,31 +1,39 @@
...
   options = {
 
     services.shellhub-agent = {
 
-      enable = mkOption {
-        type = types.bool;
-        default = false;
+      enable = mkEnableOption ''
+        Whether to enable the ShellHub Agent daemon, which allows
+        secure remote logins.
```
Comment (thiagokokada) [inline nixos/modules/services/networking/shellhub-agent.nix:17]: `mkEnableOption` already adds the "Whether to enable" prefix: https://github.com/NixOS/nixpkgs/blob/master/lib/options.nix#L97.

Comment (otavio) [reply nixos/modules/services/networking/shellhub-agent.nix]: thanks; fixed.

File: nixos/modules/services/networking/shellhub-agent.nix [hunk 3]
```diff
@@ -74,9 +79,13 @@ in {
...
 
-      environment.SERVER_ADDRESS = cfg.server;
-      environment.PRIVATE_KEY = cfg.privateKey;
-      environment.TENANT_ID = cfg.tenantId;
+      environment = {
+        SHELLHUB_SERVER_ADDRESS = cfg.server;
+        SHELLHUB_PRIVATE_KEY = cfg.privateKey;
+        SHELLHUB_TENANT_ID = cfg.tenantId;
+        SHELLHUB_KEEPALIVE_INTERVAL = toString cfg.keepAliveInterval;
+        SHELLHUB_PREFERRED_HOSTNAME = cfg.preferredHostname;
```
Comment (thiagokokada) [inline nixos/modules/services/networking/shellhub-agent.nix:87]: Considering the name isn't this option optional? If so, maybe consider setting this value to default for `null` and using `lib.mkIf (cfg.preferredHostname != null) cfg.preferredHostname` here instead.

File: nixos/modules/services/networking/shellhub-agent.nix [hunk 4]
```diff
@@ -85,7 +94,6 @@ in {
         ExecStart = "${cfg.package}/bin/agent";
       };
     };
-
-    environment.systemPackages = [ cfg.package ];
```
Comment (thiagokokada) [inline nixos/modules/services/networking/shellhub-agent.nix:89]: Is there no reason to add the package to the environment, e.g.: so the user can run the commands to check the agent status? If yes, I would keep the package on `environment.systemPackages`.

File: nixos/modules/services/networking/shellhub-agent.nix (hunk 3 above)
Comment (otavio) [reply nixos/modules/services/networking/shellhub-agent.nix:88]: It does allow it being empty; I changed it to not require it being set.

File: nixos/modules/services/networking/shellhub-agent.nix (hunk 4 above)
Comment (otavio) [reply nixos/modules/services/networking/shellhub-agent.nix:89]: Not currently.

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**